### PR TITLE
Minimal modifications to getblocktemplate

### LINKED ISSUE #1424: Ensure that GetBlockTemplate functions as intended
We have made various changes to the block header and the proof-of-work, but we haven't checked that `GetBlockTemplate` still functions correctly (as mentioned in #826). A few people in the community Slack have started trying to use it and are encountering the responses `inconclusive` followed by `duplicate-inconclusive` when submitting blocks. We should find and fix any issues with GBT, and specify any differences from BIP-22.

### PR DESCRIPTION
**Pull request by str4d**
A simpler alternative to #1435 that ensures Zcash GBT will remain compatible with BIP 22.

Closes #1424

### CODE REVIEW

**Comment by str4d** [comment]
@luke-jr as the motivator for this simpler PR, please review.

**Comment by str4d** [comment]
@ampyx @artlav you will both want to review this too, as this may affect you. tl;dr Zcash won't launch with `coinbasevalue` support because doing so before publishing a ZIP would very likely make Zcash GBT incompatible with BIP 22.

**Comment by daira** [comment]
ut(ACK+cov). We need to do the release within a few hours but if we get more feedback before then, we'll take it into account.

File: src/rpcmining.cpp [hunk 1]
```diff
@@ -585,7 +593,17 @@ Value getblocktemplate(const Array& params, bool fHelp)
         entry.push_back(Pair("fee", pblocktemplate->vTxFees[index_in_template]));
         entry.push_back(Pair("sigops", pblocktemplate->vTxSigOps[index_in_template]));
```
Comment (luke-jr) [inline src/rpcmining.cpp]: I don't think fee/sigops are populated for the coinbasetxn.

Comment (daira) [inline src/rpcmining.cpp]: So these should just be commented out for now?

Comment (daira) [inline src/rpcmining.cpp]: According to https://github.com/bitcoin/bips/blob/master/bip-0022.mediawiki#transactions-object-format , absent ``"fee"`` or ``"sigops"`` keys mean "unknown".

**Comment by ampyx** [comment]
I would agree with the current modifications, txCoinbase includes all necessary information to construct the coinbase: founders and miners reward txes. Great pull!

File: src/rpcmining.cpp (hunk 1 above)
Comment (str4d) [inline src/rpcmining.cpp]: Yes they are:

- Fee: https://github.com/zcash/zcash/blob/master/src/miner.cpp#L354
- Sigops: https://github.com/zcash/zcash/blob/master/src/miner.cpp#L369

**Comment by daira** [comment]
ut(ACK+cov). Thanks for everyone's review! @zkbot r+

**Comment by daira** [comment]
```
=== Running testscript getblocktemplate.py ===
  File "/home/admin/bbs/zcash/build/qa/rpc-tests/test_framework/test_framework.py", line 118, in main
    self.run_test()
  File "/home/admin/bbs/zcash/build/qa/rpc-tests/getblocktemplate.py", line 51, in run_test
    assert(len(tmpl['noncerange']) == 8)
  Initializing test directory /tmp/test4uGnA_
  Assertion failed: 
  Stopping nodes
  Cleaning up
  Failed
!!! FAIL: getblocktemplate.py !!!
```

Investigating.

**Comment by artlav** [comment]
On the test fail:
I don't know python, so this might be a stupid question, but does len(tmpl['noncerange']) return the length of the represented integer value, or a length of the hex string from JSON?

I've had a few brain farts to that effect while making a miner, of assuming a binary length of a hex string and vice versa, and that line reminded me of them.

**Comment by str4d** [comment]
Yeah, the problem is it should count in hex, so it's 2 32-bit integers in hex, which is 16 bytes.

**Comment by daira** [comment]
Checking whether that fixes it locally.

**Comment by str4d** [comment]
Pushed that as a fix, so once you've verified locally you can merge.

**Comment by daira** [comment]
I'll just @zkbot r+ it now to save time.

**Comment by daira** [comment]
ACK bc54cf0.

**Pull request merged**